File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultVoice.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultVoice extends InlineQueryResult<InlineQueryResultVoice> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String voice_url;
    private String title;

    private String caption;
    private String parse_mode;
    private Integer voice_duration;

    public InlineQueryResultVoice(String id, String voiceUrl, String title) {
        super("voice", id);
        this.voice_url = voiceUrl;
        this.title = title;
    }

    public InlineQueryResultVoice caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultVoice voiceDuration(Integer voiceDuration) {
        this.voice_duration = voiceDuration;
        return this;
    }

    public InlineQueryResultVoice parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultVoice.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultVoice(id: String, private val voice_url: String, private val title: String) : InlineQueryResult<InlineQueryResultVoice>("voice", id), Serializable {

    private var caption: String? = null
    private var parse_mode: String? = null
    private var voice_duration: Int? = null

    fun caption(caption: String): InlineQueryResultVoice {
        this.caption = caption
        return this
    }

    fun voiceDuration(voiceDuration: Int?): InlineQueryResultVoice {
        this.voice_duration = voiceDuration
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultVoice {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
